git_trac/releasemgr/app.py

```diff
@@ -305,12 +348,13 @@ def describe(self):
         self.git.fetch('trac', 'develop')
         return self.git.log('--oneline', '--first-parent', 'FETCH_HEAD~..HEAD')
 
-    def merge_all(self, limit=0, milestone=None):
+    def merge_all(self, limit=0, milestone=None, statuses=None, patchbot_statuses=None):
         """
         Merge all tickets that are ready
         """
         milestone = self._normalize_milestone(milestone)
-        tickets = self._get_ready_tickets(milestone=milestone)
+        tickets = self._get_ready_tickets(milestone=milestone, statuses=statuses,
+                                          patchbot_statuses=patchbot_statuses)
```

Comment:
Can we also pass the limit into _get_ready_tickets and have it only check the patchbot status for the limited set and not all tickets? Or move the patchbot status check into the merge_all main loop over tickets. Merging is slow enough as it is since trac xmlrpc has quite a bit latency.

Reply:
I'll make it a generator